tests/e2e/tests/test_metrics_server.py

```diff
@@ -0,0 +1,72 @@
+#
+# Copyright 2024 Canonical, Ltd.
+#
+import logging
+from typing import List
+
+import pytest
+from e2e_util import config, harness, util
+
+LOG = logging.getLogger(__name__)
+
+
+def test_metrics_server(instances: List[harness.Instance]):
+    if not config.SNAP:
+        pytest.fail("Set TEST_SNAP to the path where the snap is")
+
+    instance = instances[0]
+    instance.exec(["k8s", "enable", "metrics-server"])
+
+    LOG.info("Waiting for metrics-server pod to show up...")
+    util.stubbornly(retries=15, delay_s=5).on(instance).until(
+        lambda p: "ck-metrics-server" in p.stdout.decode()
```

Comment:
so this can be `metrics-server` here as well. also, since this is JSON already, maybe we could parse the JSON instead?

Reply:
This is a very basic check to see if a pod shows up, and is done similarly in places. If we want to parse these from JSON I think we can create a card/todo for revisiting these.